rust/candid_common/src/lib.rs

```diff
@@ -0,0 +1,23 @@
+/// Hash function for [Symbolic Field Ids](https://github.com/dfinity/candid/blob/master/spec/Candid.md#shorthand-symbolic-field-ids)
+#[inline]
+pub fn idl_hash(id: &str) -> u32 {
```

Comment:
It feels a bit overkill to have a crate that just contains the hash function. The hash function is not likely to change in the future, so duplicating the code in the derive crate is not a concern.

Reply:
Agreed, this would have been more useful if the original design had panned out, since the end design would have had added a `candid_type` crate in addition that would have also depended on this new `candid_common`. As it stands, yeah this is pretty overkill.